src/main/kotlin/com/sourcegraph/cody/agent/protocol/TextDocument.kt

```diff
@@ -14,9 +14,12 @@ private constructor(
 
     @JvmStatic
     @JvmOverloads
-    fun fromPath(path: String, content: String? = null, selection: Range? = null): TextDocument {
-      val uri = Paths.get(path).toUri().toString()
-      val rfc3986Uri = Rfc3986UriEncoder.encode(uri)
+    fun fromPath(
+        file: VirtualFile,
+        content: String? = null,
+        selection: Range? = null
+    ): TextDocument {
+      val rfc3986Uri = Rfc3986UriEncoder.encode(file.url)
```

Comment:
to be verified on windows

Reply:
I checked on windows, seems to be working just fine.